Sources/StreamChat/Workers/CurrentUserUpdater.swift

```diff
@@ -71,31 +71,34 @@ class CurrentUserUpdater: Worker {
                 try session.saveCurrentDevice(deviceId)
             }) { completion?($0) }
         }
-        
-        // We already have the device saved
-        if let currentUserDTO = database.viewContext.currentUser,
-           currentUserDTO.devices.first(where: { $0.id == deviceId }) != nil {
-            saveCurrentDevice(deviceId: deviceId, completion: completion)
-            return
-        }
-        apiClient
-            .request(
-                endpoint: .addDevice(
-                    userId: currentUserId,
-                    deviceId: deviceId,
-                    pushProvider: pushProvider,
-                    providerName: providerName
-                ),
-                completion: { result in
-                    if let error = result.error {
-                        log.debug("Device token \(deviceId) failed to be registered on Stream's backend.\n Reason: \(error.localizedDescription)")
-                        completion?(error)
-                        return
+
+        let backgroundContext = database.backgroundReadOnlyContext
+        backgroundContext.perform { [weak self] in
+            if let currentUserDTO = backgroundContext.currentUser,
+               currentUserDTO.devices.first(where: { $0.id == deviceId }) != nil {
+                saveCurrentDevice(deviceId: deviceId, completion: completion)
+                return
+            }
+
+            self?.apiClient
```

Comment:
We should not be starting an API request from a background thread 👀 . It would be better to get the model, and then switch back to the (presumably) main thread

Reply:
Why not? 🤔

Reply:
The request call with be added to the operation queue, so there's no point here to change it to another thread since it will be instantly changed again to the operation queue in the chat client.